prow/config/config.go

```diff
@@ -694,11 +694,12 @@ type DefaultDecorationConfigEntry struct {
 	Config *prowapi.DecorationConfig `json:"config,omitempty"`
 }
 
-// TODO(mpherman): Make a Matcher struct embedded in both ProwJobDefaultEntry and DefaultDecorationConfigEntry
-func matches(orgRepo, givenCluster, repo, cluster string) bool {
-	repoMatch := orgRepo == "" || orgRepo == "*" || orgRepo == repo || orgRepo == strings.Split(repo, "/")[0]
+// TODO(mpherman): Make a Matcher struct embedded in both ProwJobDefaultEntry and
+// DefaultDecorationConfigEntry and DefaultRerunAuthConfigEntry
+func matches(givenOrgRepo, givenCluster, orgRepo, cluster string) bool {
+	orgRepoMatch := givenOrgRepo == "" || givenOrgRepo == "*" || givenOrgRepo == strings.Split(orgRepo, "/")[0] || givenOrgRepo == orgRepo
```

Comment:
notice that the order of `givenOrgRepo == strings.Split(orgRepo, "/")[0]` and `givenOrgRepo == orgRepo` are reversed here, does it matter? if so please add a comment

Reply:
The order here does not matter. It was switched for readability purposes, as the logic order checks from lowest specificity to highest specificity. In this particular case, after the change, Org is checked and then OrgRepo.

Reply:
ack